tests/yam/migration/setup_upgrade_env.pm

```diff
@@ -0,0 +1,41 @@
+# SUSE's openQA tests
+#
+# Copyright 2024 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Set environmental variables which differ between the products involved in the upgrade
+# with the possibility to restore it later.
+# Maintainer: QE YaST and Migration (QE Yam) <qe-yam at suse de>
+
+use base "opensusebasetest";
+use strict;
+use warnings;
+use testapi;
+use migration 'reset_consoles_tty';
+
+sub run {
+    # Read product variables of the product to migrate from/to
+    my $version = get_var('VERSION_UPGRADE_FROM', get_var('VERSION_UPGRADE_TO'));
+    my $scc_addons =
+      get_var('SCC_ADDONS_UPGRADE_FROM',
+        get_var('SCC_ADDONS_UPGRADE_TO',
+            get_var('SCC_ADDONS')));
+
+    # Save the original value of the variables in order to restore it later if needed
+    set_var('VERSION_ENV', get_var('VERSION'));
+    set_var('SCC_ADDONS_ENV', get_var('SCC_ADDONS'));
+
+    # Change variables to the other version that we want to migrate from/to
+    set_var('VERSION', $version);
+    record_info('VERSION', 'VERSION=' . get_var('VERSION'));
+    set_var('SCC_ADDONS', $scc_addons);
+    record_info('SCC_ADDONS', 'SCC_ADDONS=' . get_var('SCC_ADDONS'));
```

Comment:
Resetting the SCC_ADDONS here is only required for the job migrating `SLES12SP5 to SLES15SPx+MU,` for other jobs this is not needed. For example: https://openqa.suse.de/tests/14883588#step/setup_upgrade_env/2
For testsuite `migr_sles15sp2_sp3`, the SCC_ADDONS are the same before and after migration (except to dropped ltss, btw, I think we could remove ltss form SCC_ADDONS, it is not registered in this job). So do you think need to skip this 'set_var' for these jobs? 
And same with `[migr_sles12sp5_15sp4](https://openqa.suse.de/tests/14888143#)` for example, this testsuite is to migrate SLES12SP5+MU to SLES15SP4, no need to add 15SP4's MU repos during migration, so no need to reset SCC_ADDONS values too.

Reply:
Yes, it is fine at the moment, and it doesn't affect the testing so it doesn't hurt.